auth-server/src/services/auth.ts

```diff
@@ -636,42 +638,53 @@ export default class AuthService {
 
 		const user = await getUserFromUserId(undoToken.user_id);
 
-		return { updatedToken: await this.getSignedToken(user), email: user.email };
+		return { email: user.email, updatedToken: await this.getSignedToken(user) };
 	}
 
-	private async getSignedToken({ id, name, username, email, email_verified }): Promise<string> {
+	private async getSignedToken ({ email, email_verified, id, name, username }: User): Promise<string> {
+		if (!privateKey) {
+			const key = process.env.NODE_ENV === 'test' ? 'JWT_PRIVATE_KEY_TEST' : 'JWT_PRIVATE_KEY';
+			throw new ForbiddenError(`${key} not set. Aborting.`);
+		}
+
+		if (!passphrase) {
+			const key = process.env.NODE_ENV === 'test' ? 'JWT_KEY_PASSPHRASE_TEST' : 'JWT_KEY_PASSPHRASE';
+			throw new ForbiddenError(`${key} not set. Aborting.`);
+		}
+
 		const allowedRoles: Role[] = [Role.USER];
 		let currentRole: Role = Role.USER;
 
 		const addresses = await getAddressesFromUserId(id);
 		const notification = await getNotificationPreferencesFromUserId(id);
 
 		// if our user is the proposal bot, give additional role.
-		if (id == process.env.PROPOSAL_BOT_USER_ID) { // eslint-disable-line
+		if (id === Number(process.env.PROPOSAL_BOT_USER_ID)) { // eslint-disable-line
```

Comment:
I guess we should check for the env to be there as well.

Reply:
done